File: Dart/palindrome-number.dart
class Solution {
  bool isPalindrome(int x) {
    // Negative numbers are not palindrome
    if (x < 0) return false;
    // Numbers ending with 0 (but not 0 itself) are not palindrome
    if (x % 10 == 0 && x != 0) return false;
    // Reverse the second half of the number
    int reversedHalf = 0;
    int num = x;
    // while the number is greater than the reversed half
    while (num > reversedHalf) {
      reversedHalf = reversedHalf * 10 + num % 10;
      num ~/= 10; // integer division
    }
    // Compare the first half and the reversed second half
    return num == reversedHalf || num == reversedHalf ~/ 10;
  }
}

void main() {
  var sol = Solution();
  print(sol.isPalindrome(121)); // true
  print(sol.isPalindrome(-121)); // false
  print(sol.isPalindrome(10)); // false
}

File: Javascript/palindrome-number.js
// Function to check if a number is a palindrome
var isPalindrome = function (x) {
    // Check if the number is negative or ends with a zero (except for zero itself)
    if (x < 0) return false;
    // if the number is divisible by 10, it cannot be a palindrome
    if (x % 10 === 0 && x !== 0) return false;

    // Initialize variables for the original number and its reverse
    let reversedHalf = 0;
    let num = x;
    // while loop to reverse the digits of the number
    while (num > reversedHalf) {
        reversedHalf = reversedHalf * 10 + (num % 10);
        num = Math.floor(num / 10);
    }
    // return the result of comparing the original number with its reverse
    return num === reversedHalf || num === Math.floor(reversedHalf / 10);
};

console.log(isPalindrome(12345)); // Output: false
console.log(isPalindrome(121)); // Output: true
console.log(isPalindrome(1221)); // Output: true